HCCGo/app/js/jobSubmissionCtrl.js

```diff
@@ -55,6 +60,39 @@ jobSubmissionModule.controller('jobSubmissionCtrl', ['$scope', '$log', '$timeout
     editor.setValue($scope.job.commands);
   }
 
+  $scope.chkLocation = function() {
+    //$timeout(function() {
+      if($scope.job.location) {
+        if($scope.job.location.search(/^\w*\.\w*$/)!=-1) {
+          $scope.job.changeLocation = true;
+        }
+        else {
+          $scope.job.changeLocation = false;
+        }
+      }
+    //},1000)
+  }
+  $scope.chkError = function() {
+      if($scope.job.error) {
+        if ($scope.job.error.search(/^\w*\.\w*$/)!=-1) {
+          $scope.job.changeError = true;
+        }
+        else {
+          $scope.job.changeError = false;
+        }
+      }
+  }
+  $scope.chkOutput = function() {
+      if($scope.job.output) {
+        if ($scope.job.output.search(/^\w*\.\w*$/)!=-1) {
+          $scope.job.changeOutput = true;
+        }
+        else {
+          $scope.job.changeOutput = false;
+        }
+      }
+  }
+
```

Comment:
What if you make 1 function.  Pass in the variable and something else.  So the `ng-change` for the error file would look like:

    ... ng-change="chkDir(job.error, 'error')"

Then, then function would look something like:

```javascript
$scope.chkDir = function(dir, identifier) {
  if (dir) {
    if (dir.search(/^\w*\.\w*$/)!=-1) {
      $scope.job[identifier]["changedir"] = true;
    }
    else {
      $scope.job[identifier]["changedir"] = false;
    }
  }
}
```

Then, the `span`s would look like:

```html
 ... ng-show="job.error.changedir" ...
```

Does this make sense?